python/ray/autoscaler/_private/_azure/node_provider.py

```diff
@@ -65,7 +66,17 @@ class AzureNodeProvider(NodeProvider):
 
     def __init__(self, provider_config, cluster_name):
         NodeProvider.__init__(self, provider_config, cluster_name)
-        subscription_id = provider_config["subscription_id"]
+        subscription_id = provider_config.get("subscription_id")
+        if subscription_id is None:
+            # Get subscription from logged in azure profile
+            # if it isn't provided in the provider_config
+            # so operations like `get-head-ip` will work
+            subscription_id = get_cli_profile().get_subscription_id()
+            provider_config["subscription_id"] = subscription_id
```

Comment:
are there any negative side-effects from updating the in-memory representation of the parsed user config in this way?

Reply:
updated